### PR TITLE
[stdlib] Remove a bunch of declarations marked as obsoleted in 4

### PR DESCRIPTION
**Pull request by moiseev**

### CODE REVIEW

**Comment by moiseev** [comment]
@swift-ci Please test

**Comment by moiseev** [comment]
@swift-ci Please test

File: test/api-digester/Outputs/stability-stdlib-abi.swift.expected [hunk 1]
```diff
@@ -14,6 +14,45 @@ Var _StringObject.undiscriminatedObjectRawBits has been removed
...
+Func Int64.<<(_:_:) has been removed
+Func Int64.<<=(_:_:) has been removed
+Func Int64.>>(_:_:) has been removed
+Func Int64.>>=(_:_:) has been removed
+Func Int64.toUIntMax() has been removed
+Func Int8.<<(_:_:) has been removed
+Func Int8.<<=(_:_:) has been removed
+Func Int8.>>(_:_:) has been removed
+Func Int8.>>=(_:_:) has been removed
+Func Int8.toUIntMax() has been removed
```
Comment (xedin) [inline test/api-digester/Outputs/stability-stdlib-abi.swift.expected]: \o/

**Comment by moiseev** [comment]
@swift-ci please smoke test compiler performance

**Comment by moiseev** [comment]
@swift-ci Please test

**Comment by moiseev** [comment]
@swift-ci Please test macOS platform

**Comment by moiseev** [comment]
@swift-ci Please test macOS platform

**Comment by moiseev** [comment]
@swift-ci Please test

**Comment by moiseev** [comment]
@swift-ci Please test

File: stdlib/public/core/Integers.swift [hunk 2]
```diff
@@ -3720,24 +3581,10 @@ extension SignedInteger {
...
   public static func _maskingSubtract(_ lhs: Self, _ rhs: Self) -> Self {
     fatalError("Should be overridden in a more specific type")
   }
-
-  @_transparent
-  @available(swift, obsoleted: 4.0,
-      message: "Please use 'FixedWidthInteger' instead of 'SignedInteger' to get '&-' in generic code.")
-  public static func &- (lhs: Self, rhs: Self) -> Self {
-    return _maskingSubtract(lhs, rhs)
-  }
```
Comment (jrose-apple) [inline stdlib/public/core/Integers.swift]: Some of these things are here to help when people type the wrong thing. Are you sure it's worth removing them?

Comment (moiseev) [reply stdlib/public/core/Integers.swift]: Yeah, that was when `SignedInteger` implied a fixed-width-integer-ness, which has not been the case for more than a year no, so I'd say, yes.

**Comment by xedin** [comment]
:shipit:

**Comment by moiseev** [comment]
@swift-ci Please test

**Comment by moiseev** [comment]
https://github.com/apple/swift-lldb/pull/1255
@swift-ci Please smoke test

**Comment by moiseev** [comment]
@swift-ci Please smoke test

**Comment by moiseev** [comment]
@swift-ci Please test macOS platform

**Comment by benrimmington** [comment]
Should the [`LazySequenceProtocol.flatMap(_:)`](https://github.com/apple/swift/blob/3c48f1ca1421a325fb70a45037f1aaa7d60e3969/stdlib/public/core/MigrationSupport.swift#L183-L204) also be obsoleted?

**Comment by moiseev** [comment]
> Should the LazySequenceProtocol.flatMap(_:) also be obsoleted?

Go for it. `obsoleted: 5.1` seems to be a reasonable guess for a next version.

As for the quicklookables, the majority of that stuff is already marked as obsoleted in Swift 5.0, the rest for some reason wasn't. Do you mind trying to mark it as obsolete in 5.1 and see what breaks?
/cc @airspeedswift

**Comment by benrimmington** [comment]
@moiseev My current iMac isn't supported by macOS Mojave, so I won't be able to build and test the above changes (to flatMap and quicklookables) by March 18th.

**Comment by moiseev** [comment]
It doesn't look like a pressing issue at the moment. All the APIs we're talking about are marked as deprecated now and the don't have to be obsoleted in 5.0 anyway. Besides, CI can help figure out what breaks :-)

**Comment by moiseev** [comment]
Let's try that `flatMap` first, shall we? https://github.com/apple/swift/pull/22817

**Pull request merged**